### DISCUSSION TITLE
Supress multiple Suricata alarms

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#4640] [Unsupported Versions]
Hello!

I learned how to supress Suricata alarms like this in /opt/so/saltstack/local/pillar/minions/my-sensor.sls:

 thresholding:
    sids:
      1234567:
        -supress:
          gen_id: 1
          track: by_dst
          ip: 1.2.3.4

How does the syntax look like, if I want to supress the alarm for multiple IPs? In the Suricata docs I found something like [IP1,IP2,...], but that doesn't seem to work. Thanks in advance!

Cheers, Ben

**Comment by gcsrellis** [comment]
If it's a group of IP's that trigger similar alarms you can create an address-group variable for suricata and use that for the ip in thresholding.

suricata:
&nbsp; &nbsp; config:
&nbsp; &nbsp; &nbsp; &nbsp; vars:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; address-groups:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; CUSTOM_VARIABLE: '[192.168.1.1,8.8.8.8,1.1.1.1]'

thresholding:
&nbsp; &nbsp; sids:
&nbsp; &nbsp; &nbsp; &nbsp; 2027201:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; - suppress:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; gen_id: 1
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; track: by_dst
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; ip: $CUSTOM_VARIABLE




Have you also tried enclosing the ip like below

thresholding:
&nbsp; &nbsp; sids:
&nbsp; &nbsp; &nbsp; &nbsp; 2027201:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; - suppress:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; gen_id: 1
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; track: by_dst
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; ip: '[1.1.1.1,2.2.2.2,3.3.3.3]'

**Reply by ben-sec** [reply]
According to your syntax I missed the **'** around the [1.1.1.1]. I will give that a try. However, I found the following statement in the current SO documentation:

"Please note that Suricata 6 has a 64-character limitation on the IP field in a threshold."

And it is suggesting the following:

thresholding:
  sids:
    2012454:
      - suppress:
          gen_id: 1
          track: by_dst
          ip: 1.1.1.1,2.2.2.2,3.3.3.3,4.4.4.4,5.5.5.5,6.6.6.6,7.7.7.7,8.8.8.8
      - suppress:
          gen_id: 1
          track: by_dst
          ip: 9.9.9.9,10.10.10.10,11.11.11.11

That would be my next try.

**Reply by gcsrellis** [reply]
Awesome, looks like you got it.  You could try the variable if you have more than 64 characters too.  It may or may not have the same limitation.

**Reply by ben-sec** [reply]
I tried the workaround regarding the 64 char limit as described in the SO documentation without success. Therefore I have opened a [new discussion](https://github.com/Security-Onion-Solutions/securityonion/discussions/4693).
